### PR TITLE
Immutable.js overhaul

### PR DESCRIPTION
**Pull request by fourwallsstudio**

### CODE REVIEW

File: frontend/components/footer/index.jsx [hunk 1]
```diff
@@ -20,14 +20,18 @@ export class Footer extends Component {
   }
 
   handleAddTask = (values) => {
-    values['priority'] = this.props.newPriority;
-    this.props.createTask(values);
+    values = values.set('priority', this.props.newPriority);
+    this.props.createTask(values.toJS());
```
Comment (pseuyi) [inline frontend/components/footer/index.jsx:24]: why `toJS()` here?

File: frontend/components/task_modal/index.jsx [hunk 2]
```diff
@@ -67,12 +67,12 @@ const Button = styled.button`
 const TaskModal = (props) => {
 
   const handleSubmitEditTask = (values) => {
-    props.editTask(values);
+    props.editTask(values.toJS());
```
Comment (pseuyi) [inline frontend/components/task_modal/index.jsx:70]: is the reducer not expecting an immutable data structure?

File: frontend/reducers/task_reducer.js [hunk 3]
```diff
@@ -1,31 +1,32 @@
...
 
 const taskReducer = (state = defaultState, action) => {
   switch (action.type) {
     case actions.RECEIVE_TASKS:
-      return {
-        tasksByIds: state.tasksByIds.merge(action.payload.entities.tasks),
-        ids: state.ids.concat(action.payload.result.tasks),
-      };
+      return state
+        .set('tasksByIds', fromJS(action.payload.entities.tasks))
```
Comment (pseuyi) [inline frontend/reducers/task_reducer.js:14]: payload is js?

File: frontend/components/footer/index.jsx (hunk 1 above)
Comment (fourwallsstudio) [reply frontend/components/footer/index.jsx:24]: createTask is an api action

File: frontend/components/task_modal/index.jsx (hunk 2 above)
Comment (fourwallsstudio) [reply frontend/components/task_modal/index.jsx:70]: editTask is also an api action

File: frontend/reducers/task_reducer.js (hunk 3 above)
Comment (fourwallsstudio) [reply frontend/reducers/task_reducer.js:14]: yes but maybe massageData should convert payload to Immutable.js instead of in reducers?

**Pull request merged**